src/main/java/org/spoken_tutorial/health/elasticsearch/contentfile/ContentsfromFile.java

```diff
@@ -19,21 +26,103 @@
 @Component
 public class ContentsfromFile {
 
+    private static final Logger logger = LoggerFactory.getLogger(ContentsfromFile.class);
+
     @Autowired
     private Config config;
 
+    private boolean containsOneSlash(String str) {
+        return str.chars().filter(ch -> ch == '/').count() == 1;
+    }
+
+    private boolean containsMoreThanTwoSlashes(String str) {
+        return str.chars().filter(ch -> ch == '/').count() > 2;
+    }
+
     public String extractContent(final Parser parser, final String filePath)
             throws IOException, SAXException, TikaException {
 
-        long start = System.currentTimeMillis();
+        String content = "";
+        List<String> allowedExtensions = Arrays.asList(config.ALLOWED_EXTENTIONS.split(","));
 
         BodyContentHandler handler = new BodyContentHandler(10000000);
         Metadata metadata = new Metadata();
         Path path = Paths.get(config.BASE_PATH, filePath);
-        InputStream content = Files.newInputStream(path);
-        parser.parse(content, handler, metadata, new ParseContext());
+        InputStream inputStream = Files.newInputStream(path);
+
+        if (allowedExtensions.contains(filePath.substring(filePath.lastIndexOf(".") + 1))) {
+
+            if (filePath.toLowerCase().endsWith(".zip")) {
+                ZipInputStream zis = new ZipInputStream(inputStream);
+                ZipEntry zipEntry;
+                int count = 0;
+                while ((zipEntry = zis.getNextEntry()) != null) {
+                    if (!zipEntry.isDirectory() && !zipEntry.getName().toLowerCase().endsWith(".zip")
```

Comment:
you require abother function here. check https://www.baeldung.com/java-compress-and-uncompress for ideas on how to handle subdirectories.

the other function will handle one zipentry. if a directory, it will call itself with the new zipentry with depth increased by 1. if depth >= DEPTH_LIMIIT, it will return. initial value of depth is 0.